### PR TITLE
added input for set the key name

### PR DESCRIPTION
**Pull request by Maxstgt**
@pldespaigne I don't think we need an interface here for the KeyName. Cause the address won't be used in this component. Otherwise we have to input it, just to output it again. The mapping could happen in the parent component.

### CODE REVIEW

File: src/app/wallet/wallet-password-form/wallet-password-form.component.ts [hunk 1]
```diff
@@ -13,9 +14,14 @@ import {
   changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class WalletPasswordFormComponent {
+  public setKeyName: string;
+  public setKeyNameFormGroup = new FormGroup({
```
Comment (GrandSchtroumpf) [inline src/app/wallet/wallet-password-form/wallet-password-form.component.ts]: You shouldn't have a FormGroup with only one field.

Comment (GrandSchtroumpf) [reply src/app/wallet/wallet-password-form/wallet-password-form.component.ts]: You could merge this FormGroup with the password one : 
```
public form = new FormGroup({
  keyName: new FormControl('', [Validators.required]),
  password: new FormControl('', [Validators.required]),
})
```

File: src/app/wallet/wallet-password-form/wallet-password-form.component.ts [hunk 2]
```diff
@@ -13,9 +14,14 @@ import {
...
+    nameOfKey: new FormControl(this.setKeyName, [Validators.required])
```
Comment (GrandSchtroumpf) [inline src/app/wallet/wallet-password-form/wallet-password-form.component.ts]: Where `this.setKeyName` comes from ?

File: src/app/wallet/wallet-password-form/wallet-password-form.component.html [hunk 3]
```diff
@@ -19,7 +19,8 @@ <h1>Unlock your key</h1>
         China. It could also be lots of other people. It also could be some
         wordsmith sitting on their bed that weights 400 pounds. Ok?
       </p>
-      <mat-form-field>
+      <mat-form-field [formGroup]="setKeyNameFormGroup">
+        <input matInput [formControlName]="nameOfKey" />
```
Comment (GrandSchtroumpf) [inline src/app/wallet/wallet-password-form/wallet-password-form.component.html]: Here the `formControlName` should evaluate a string, and not a javascript value so you should do : 
```
 <input matInput formControlName="nameOfKey" />
```

File: src/app/wallet/wallet-password-form/wallet-password-form.component.html [hunk 4]
```diff
@@ -19,7 +19,8 @@ <h1>Unlock your key</h1>
         China. It could also be lots of other people. It also could be some
         wordsmith sitting on their bed that weights 400 pounds. Ok?
       </p>
-      <mat-form-field>
+      <mat-form-field [formGroup]="setKeyNameFormGroup">
```
Comment (GrandSchtroumpf) [inline src/app/wallet/wallet-password-form/wallet-password-form.component.html]: You should have one mat-form-field / input. The `[formGroup]` should be on a `<form>` tag around

File: src/app/wallet/wallet-password-form/wallet-password-form.component.ts (hunk 2 above)
Comment (Maxstgt) [reply src/app/wallet/wallet-password-form/wallet-password-form.component.ts]: from line 17

**Pull request closed without merging**